lib/src/util/mixins.dart

```diff
@@ -819,10 +819,13 @@ mixin MessageRespondMixin implements InteractiveMixin {
     if (builder.replyId == null) {
       builder.replyId = message.id;
 
-      // Calling [AllowedMentions.allow] here will only change anything if [level.mention] is
-      // non-null, in which case we want to change it. Otherwise, this does nothing.
-      builder.allowedMentions!.parse =
-          {...(builder.allowedMentions!.parse ?? <String>[]), 'replied_user'}.toList();
+      if (level.mention == true) {
+        builder.allowedMentions =
+            (builder.allowedMentions ?? AllowedMentions()) | AllowedMentions(repliedUser: true);
+      } else if (level.mention == false) {
+        builder.allowedMentions =
+            (builder.allowedMentions ?? AllowedMentions()) & AllowedMentions(repliedUser: false);
+      }
```

Comment:
is == needed? cant if be if(level.mention) {} else {} ?

Reply:
`mention` is nullable and if null, we want to leave `allowedMentions` as-is, so we only check for non-null values.

Reply:
Maybe something like this would be less verbose
```dart
if (level.mention case bool m) {
  final allowedMentions = builder.allowedMentions ?? AllowedMentions();
  final mentions = AllowedMentions(repliedUser: m);
  builder.allowedMentions = m ? (allowedMentions & mentions) : (allowedMentions | mentions);
}
```

Reply:
@Rapougnac sugesstion is slightly better. WOuld name m variable a little bit better - `levelMention`